random_fields/geostatistical_cpt_interpretation.py

```diff
@@ -0,0 +1,682 @@
+        """
+
+        self.data = np.sort(data)
+        self.n = len(data)
+        self.lognormal = lognormal
+
+        # if `max_value` and `min_value` are not given: add a 1% margin on the data
+        if not min_value:
+            min_value = self.data.min() - (self.data.max() - self.data.min()) / 100
+            min_ln = np.log(self.data.min()) - (np.log(self.data.max()) - np.log(self.data.min())) / 100
+        else:
+            min_ln = np.log(min_value)
+        if not max_value:
+            max_value = self.data.max() + (self.data.max() - self.data.min()) / 100
+            max_ln = np.log(self.data.max()) + (np.log(self.data.max()) - np.log(self.data.min())) / 100
+        else:
+            max_ln = np.log(max_value)
+
+        self.z = st.norm.ppf(np.linspace(.5 / self.n, 1 - .5 / self.n, self.n))
+
+        # adds minimum and maximum to data-set
+        x_points = np.hstack([np.float64(min_value), self.data, np.float64(max_value)])
+        x_points_ln = np.hstack([min_ln, np.log(self.data), max_ln])
+        z_points = np.hstack([-8., self.z, 8.])
+
+        # initiates internal points for interpolation
+        self._interp_z = np.linspace(-8, 8, 101)
+        self._interp_data = np.interp(self._interp_z, z_points, x_points)
+        self._interp_log_data = np.interp(self._interp_z, z_points, x_points_ln)
+
+    def plot(self,
+             title: str = "Marginal distribution transformation",
+             output_folder: Path = Path("./"),
+             output_name: str = "marginal_distribution.png",
+             x_label: str = r'$Z\sim N(0, 1)$',
+             y_label: str = r'$X$',
+             figsize: Tuple[float, float] = (6, 5),
+             show: bool = False) -> None:
+        r"""Plots the transformation model for the marginal distribution.
+
+        Args:
+            - title (str): plot title. Defaults to "Marginal distribution transformation ".
+            - output_folder (Path): output folder. Defaults to Path("./").
+            - output_name (str): output file name. Defaults to "marginal_distribution.png".
+            - x_label (str): horizontal axis label. Defaults to '\sim N(0,1)$'.
+            - y_label (str): vertical axis label. Defaults to '$X$'.
+            - figsize (Tuple): figure size. Defaults to (6, 5).
+            - show (bool): If True, shows the plot. Defaults to False.
+        """
+        plt.figure(figsize=figsize)
+        plt.plot(self.z, self.data, '.', color='gray', alpha=0.5, label='data')
+        plt.plot(self._interp_z, self._interp_data, '-', color='r', label='transformation model')
+        plt.xlabel(x_label)
+        plt.ylabel(y_label)
+        plt.legend()
+        plt.title(title)
+        plt.grid()
+        plt.savefig(os.path.join(output_folder, output_name))
+        if show:
+            plt.show()
+        plt.close()
+
+    def x_to_z(self, x: npt.NDArray[np.float64]):
```

Comment:
i find the naming of "x_to_z" confusing, as often we work with coordinates, xyz, in this case, x and z have different meaning